internal/pkg/deploy/cloudformation/stack/env.go

```diff
@@ -192,6 +197,36 @@ func (e *EnvStackConfig) Parameters() ([]*cloudformation.Parameter, error) {
 	return e.transformParameters(currParams, e.prevParams, transformEnvControllerParameters)
 }
 
+func getSecurityGroupConfig(mft *manifest.Environment) (*template.SecurityGroupConfig, error) {
+	if mft != nil && !mft.Network.VPC.SecurityGroupConfig.IsEmpty() {
+		if !mft.Network.VPC.SecurityGroupConfig.IsValid() {
+			var ingress = make([]template.SecurityGroupRules, len(mft.Network.VPC.SecurityGroupConfig.Ingress))
+			var egress = make([]template.SecurityGroupRules, len(mft.Network.VPC.SecurityGroupConfig.Egress))
```

Comment:
I mentioned the Law of Demeter in this PR: https://github.com/aws/copilot-cli/pull/3737#discussion_r929180846 what do you think of having the following method:

```go
package manifest

// EnvSecurityGroup returns the security group config if the user has set any values. 
// If there is no env security group settings, then returns nil and false.
func (mft *EnvironmentConfig) EnvSecurityGroup() (*SecurityGroupConfig, bool) 
```

Reply:
Sounds good to me I have added this method to check if env security group config exist.